### PR TITLE
[libigl] Update to v2.5.0

### PR DESCRIPTION
**Pull request by NewWheelTech**

### CODE REVIEW

**Comment by NewWheelTech** [comment]
@microsoft-github-policy-service agree company="New Wheel Technology LLC"

**Comment by dg0yt** [comment]
Without pointing at a particular modification:
The port probably had problems before this PR... The patches need consolidation. Changes of incoming and outgoing dependencies should be paired in the same patch IMO. In the installed config, `find_dependency` calls must come before creating this package's targets, so they can't depend on the targets.

**Comment by NewWheelTech** [comment]
> Without pointing at a particular modification: The port probably had problems before this PR... The patches need consolidation. Changes of incoming and outgoing dependencies should be paired in the same patch IMO. In the installed config, `find_dependency` calls must come before creating this package's targets, so they can't depend on the targets.

Thanks for taking a look at this, I would really like to have this package updated but it appears to vcpkg ports have been neglected for sometime and this is a little bit beyond my knowledge. 

Where should I add these find_dependency calls? I can merge the patches into one patch file if that is a better fit as well.

File: ports/libigl/install-extra-targets.patch [hunk 1]
```diff
@@ -12,12 +12,12 @@ index 8143963..784ac50 100644
...
-index 629896d..0b05692 100644
+index d3ff19f..60e50c7 100644
 --- a/cmake/igl/libigl-config.cmake.in
 +++ b/cmake/igl/libigl-config.cmake.in
-@@ -4,5 +4,30 @@ include(CMakeFindDependencyMacro)
- find_dependency(Eigen3)
- find_dependency(Threads)
+@@ -4,4 +4,27 @@ include(CMakeFindDependencyMacro)
+ find_dependency(Eigen3 REQUIRED)
+ find_dependency(Threads REQUIRED)
```
Comment (dg0yt) [inline ports/libigl/install-extra-targets.patch:20]: `REQUIRED` is wrong in `find_dependency`.

Comment (NewWheelTech) [reply ports/libigl/install-extra-targets.patch:20]: This isn't part of the patch, it is the existing code base. 

For some reason the libigl team thinks this is required, 

https://github.com/libigl/libigl/pull/2188/commits/82a43432a8a78fc3e16e766cd9641922dcaeedb8

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

The usage test passed on `x64-windows` (header files found):
```
libigl provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(libigl CONFIG REQUIRED)
  # note: 4 additional targets are not displayed.
  target_link_libraries(main PRIVATE igl::igl_xml igl::igl_core igl::igl_glfw igl::igl_imgui)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**